### ISSUE TITLE
Numeric input doesn't accept more than 16 digits, digits after 16 are converted to zeros!!

### ISSUE THREAD

**Issue by MOB98** [carbon-design-system/carbon-components-angular#2722]
https://github.com/carbon-design-system/carbon-components-angular/blob/6b50b7d3245d285e34f8ad52a8bf69cb6937cd27/src/number-input/number.component.ts#L188C19-L188C19

Numeric input doesn't accept more than 16 digits, digits after 16 are converted to zeros!!
Is there a way to enter value more than 16 digits without converting to zeros or exponential formula.
I think that's happen because carbon always writes value as Number(value) and Number type doesn't accept more than 16 digi

**Comment by nipun005** [comment]
Hi, We can use "_value" type as BigInt here instead of number type, which can accept more than 16 digits. 
Only the thing is that, If we will go with this approach, we will be needed to use BigInt() in all places if we wanted to perform any calculation with this number.

Suppose if we want to add -1 to this number, On code base- We needed to write- "this._value + BigInt(1)"

**Comment by MOB98** [comment]
so how I can type more than 16 digits without doing changes on the code base?
I mean, Does the current behaviour of carbon support typing more than 16 digits?
Thanks @nipun005  for your response.

**Comment by nipun005** [comment]
No, @MOB98 , Current codebase of carbon will not support more than 16 digits, 
The reason is that in all the codebase of number.component.ts file, we have used Datatype as "number" which supports upto 16 digits only,
So, at all those places we need to change datatype to "BigInt", which can fixed this issue. Only thing is that - after changing we need to do proper unit testing so that we can make sure that we changed this datatype from "number" to "BigInt" at every place of this number.component.ts file, otherwise we might get type error at that particular instance.

**Comment by MOB98** [comment]
any updates on this issue @nipun005
